src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs

```diff
@@ -61,7 +61,7 @@ private static bool ProxySupportsConnectionAuth(HttpResponseMessage response)
 
             foreach (string v in values)
             {
-                if (v == "Session-Based-Authentication")
```

Comment:
Separately, @geoffkizer, we might want to audit use of Headers and update appropriate sites to use NonValidated.

Reply:
Is the goal here to improve performance by avoiding the added processing/memory use that happens when you go through TryGetValues etc?

There are semantic differences with NonValidated. In particular if the server sent something like 

```
Proxy-Support: Session-Based-Authentication, Some-Other-Magic-Token
```

Then the current code would enumerate this as two header values and successfully match the first. In general for header processing, this is what you want (e.g. consider Connection: close handling).

If we just use NonValidated here then we would need to parse these tokens ourselves. Which we could certainly do, and we could certainly do it in a more efficient way than the current header parsing logic does (e.g. deal with spans instead of allocating strings). But I'm not sure the perf here matters much.

There could be (and almost certainly are) scenarios where this perf does matter, e.g. Connection: close. But again, it's not as simple as just using NonValidated.

In short: Yes, there are things we should look into as far as improving header access perf, but NonValidated by itself doesn't really help all that much, and we need to investigate further to determine if/how to actually do better here.

Reply:
> Is the goal here to improve performance by avoiding the added processing/memory use that happens when you go through TryGetValues etc?

Yes, that would be the goal.